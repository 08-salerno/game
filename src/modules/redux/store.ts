import { configureStore } from '@reduxjs/toolkit';
import createSagaMiddleware from 'redux-saga';
import userReducer from './slices/userSlice';
import rootSaga from './sagas/root.saga';

const sagaMiddleware = createSagaMiddleware();

const store = configureStore({
  reducer: {
    user: userReducer,
  },
  middleware: [sagaMiddleware],
});

sagaMiddleware.run(rootSaga);

export type RootState = ReturnType<typeof store.getState>
export type AppDispatch = typeof store.dispatch

export default store;
