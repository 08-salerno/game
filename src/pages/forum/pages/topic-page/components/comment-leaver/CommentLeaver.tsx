import React from 'react';
import { object, string, StringSchema } from 'yup';
import {
  Formik, Form, Field, FormikValues, FieldProps, FormikConfig,
} from 'formik';

export type CommentLeaverProps = {
  handleLeaveComment: (text: string) => Promise<void>;
};

interface CommentFormValue extends FormikValues {
  comment: string;
}

const CommentSchema = object().shape<{ comment: StringSchema }>({
  comment: string().required(''),
});

const MaxCommentLength = 1024;

const CommentLeaver: React.VFC<CommentLeaverProps> = (props) => {
  const { handleLeaveComment } = props;

  const value: CommentFormValue = {
    comment: '',
  };

  const handleSubmit: FormikConfig<CommentFormValue>['onSubmit'] = (
    values: CommentFormValue,
    { setValues },
  ): Promise<void> => handleLeaveComment(values.comment).then(() => {
    setValues({ comment: '' });
    return Promise.resolve();
  });

  return (
    <Formik
      initialValues={value}
      validationSchema={CommentSchema}
      onSubmit={handleSubmit}
    >
      {({ isSubmitting, isValid, dirty }): React.ReactElement => (
        <Form>
        {/* todo [sitnik] вынести в переиспользуемый компонент*/}
          <Field name="comment">
            {(props: FieldProps<string, CommentFormValue>): React.ReactElement => (
              <div>
                {/* eslint-disable-next-line react/jsx-props-no-spreading */}
                <textarea placeholder="Ваш комментарий" maxLength={MaxCommentLength} {...props.field} />
                <div>
                  {props.field.value.length || 0}/{MaxCommentLength}
                </div>
              </div>
            )}
          </Field>
          <div>
            <button type="submit" disabled={isSubmitting || !(dirty && isValid)}>
              Прокомментировать
            </button>
          </div>
        </Form>
      )}
    </Formik>
  );
};

export default CommentLeaver;
