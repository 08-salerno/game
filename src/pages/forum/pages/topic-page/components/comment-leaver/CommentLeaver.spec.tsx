import React from 'react';
import { render } from 'enzyme';
import { Provider } from 'react-redux';
import CommentLeaver from './CommentLeaver';
import store from '../../../../../../modules/redux/store';

it('CommentLeaver', () => {
  expect(render(<Provider store={store}>
                    <CommentLeaver handleLeaveComment={(text: string) => Promise.resolve()} />
                </Provider>)).toMatchSnapshot();
});
