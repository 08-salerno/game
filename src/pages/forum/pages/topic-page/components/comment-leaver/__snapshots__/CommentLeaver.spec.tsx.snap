// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommentLeaver 1`] = `
LoadedCheerio {
  "0": Node {
    "attribs": Object {
      "action": "#",
    },
    "children": Array [
      Node {
        "attribs": Object {},
        "children": Array [
          Node {
            "attribs": Object {
              "maxlength": "1024",
              "name": "comment",
              "placeholder": "Ваш комментарий",
            },
            "children": Array [],
            "name": "textarea",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": Node {
              "attribs": Object {},
              "children": Array [
                Node {
                  "data": "0/1024",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
              ],
              "name": "div",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": [Circular],
              "type": "tag",
              "x-attribsNamespace": Object {},
              "x-attribsPrefix": Object {},
            },
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "maxlength": undefined,
              "name": undefined,
              "placeholder": undefined,
            },
            "x-attribsPrefix": Object {
              "maxlength": undefined,
              "name": undefined,
              "placeholder": undefined,
            },
          },
          Node {
            "attribs": Object {},
            "children": Array [
              Node {
                "data": "0/1024",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "text",
              },
            ],
            "name": "div",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": Node {
              "attribs": Object {
                "maxlength": "1024",
                "name": "comment",
                "placeholder": "Ваш комментарий",
              },
              "children": Array [],
              "name": "textarea",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": [Circular],
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "maxlength": undefined,
                "name": undefined,
                "placeholder": undefined,
              },
              "x-attribsPrefix": Object {
                "maxlength": undefined,
                "name": undefined,
                "placeholder": undefined,
              },
            },
            "type": "tag",
            "x-attribsNamespace": Object {},
            "x-attribsPrefix": Object {},
          },
        ],
        "name": "div",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": Node {
          "attribs": Object {},
          "children": Array [
            Node {
              "attribs": Object {
                "disabled": "",
                "type": "submit",
              },
              "children": Array [
                Node {
                  "data": "Прокомментировать",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
              ],
              "name": "button",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "disabled": undefined,
                "type": undefined,
              },
              "x-attribsPrefix": Object {
                "disabled": undefined,
                "type": undefined,
              },
            },
          ],
          "name": "div",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": null,
          "parent": [Circular],
          "prev": [Circular],
          "type": "tag",
          "x-attribsNamespace": Object {},
          "x-attribsPrefix": Object {},
        },
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {},
        "x-attribsPrefix": Object {},
      },
      Node {
        "attribs": Object {},
        "children": Array [
          Node {
            "attribs": Object {
              "disabled": "",
              "type": "submit",
            },
            "children": Array [
              Node {
                "data": "Прокомментировать",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "text",
              },
            ],
            "name": "button",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "disabled": undefined,
              "type": undefined,
            },
            "x-attribsPrefix": Object {
              "disabled": undefined,
              "type": undefined,
            },
          },
        ],
        "name": "div",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": Node {
          "attribs": Object {},
          "children": Array [
            Node {
              "attribs": Object {
                "maxlength": "1024",
                "name": "comment",
                "placeholder": "Ваш комментарий",
              },
              "children": Array [],
              "name": "textarea",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": Node {
                "attribs": Object {},
                "children": Array [
                  Node {
                    "data": "0/1024",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "text",
                  },
                ],
                "name": "div",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": [Circular],
                "type": "tag",
                "x-attribsNamespace": Object {},
                "x-attribsPrefix": Object {},
              },
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "maxlength": undefined,
                "name": undefined,
                "placeholder": undefined,
              },
              "x-attribsPrefix": Object {
                "maxlength": undefined,
                "name": undefined,
                "placeholder": undefined,
              },
            },
            Node {
              "attribs": Object {},
              "children": Array [
                Node {
                  "data": "0/1024",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
              ],
              "name": "div",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": Node {
                "attribs": Object {
                  "maxlength": "1024",
                  "name": "comment",
                  "placeholder": "Ваш комментарий",
                },
                "children": Array [],
                "name": "textarea",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": [Circular],
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {
                  "maxlength": undefined,
                  "name": undefined,
                  "placeholder": undefined,
                },
                "x-attribsPrefix": Object {
                  "maxlength": undefined,
                  "name": undefined,
                  "placeholder": undefined,
                },
              },
              "type": "tag",
              "x-attribsNamespace": Object {},
              "x-attribsPrefix": Object {},
            },
          ],
          "name": "div",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": [Circular],
          "parent": [Circular],
          "prev": null,
          "type": "tag",
          "x-attribsNamespace": Object {},
          "x-attribsPrefix": Object {},
        },
        "type": "tag",
        "x-attribsNamespace": Object {},
        "x-attribsPrefix": Object {},
      },
    ],
    "name": "form",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": Node {
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
    },
    "prev": null,
    "type": "tag",
    "x-attribsNamespace": Object {
      "action": undefined,
    },
    "x-attribsPrefix": Object {
      "action": undefined,
    },
  },
  "_root": LoadedCheerio {
    "0": Node {
      "children": Array [
        Node {
          "attribs": Object {},
          "children": Array [
            Node {
              "attribs": Object {},
              "children": Array [],
              "name": "head",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": Node {
                "attribs": Object {},
                "children": Array [],
                "name": "body",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": [Circular],
                "type": "tag",
                "x-attribsNamespace": Object {},
                "x-attribsPrefix": Object {},
              },
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {},
              "x-attribsPrefix": Object {},
            },
            Node {
              "attribs": Object {},
              "children": Array [],
              "name": "body",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": Node {
                "attribs": Object {},
                "children": Array [],
                "name": "head",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": [Circular],
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {},
                "x-attribsPrefix": Object {},
              },
              "type": "tag",
              "x-attribsNamespace": Object {},
              "x-attribsPrefix": Object {},
            },
          ],
          "name": "html",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": null,
          "parent": [Circular],
          "prev": null,
          "type": "tag",
          "x-attribsNamespace": Object {},
          "x-attribsPrefix": Object {},
        },
      ],
      "name": "root",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-mode": "quirks",
    },
    "_root": [Circular],
    "length": 1,
    "options": Object {
      "decodeEntities": true,
      "xml": false,
    },
  },
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "xml": false,
  },
}
`;
